### PR TITLE
Modify segment file name

### PR DESCRIPTION
**Pull request by kangpinghuang**
1、modify segment file name to rowset_id + segment_number
2、add rowset init_from_pb api
3、optimize cmake file to move rowset cmake file to subdirectory of olap

### CODE REVIEW

File: be/src/olap/rowset/rowset_meta.h [hunk 1]
```diff
@@ -42,6 +42,11 @@ class RowsetMeta {
         return _deserialize_from_pb(pb_rowset_meta);
     }
 
+    virtual bool init_from_pb(const RowsetMetaPB rowset_meta_pb) {
```
Comment (chaoyli) [inline be/src/olap/rowset/rowset_meta.h]: const RowsetMetaPB&

File: be/src/olap/rowset/segment_reader.cpp [hunk 2]
```diff
@@ -39,7 +39,6 @@ SegmentReader::SegmentReader(
         const std::vector<uint32_t>& used_columns,
         const std::set<uint32_t>& load_bf_columns,
         const Conditions* conditions,
-        const std::vector<ColumnPredicate*>* col_predicates,
```
Comment (chaoyli) [inline be/src/olap/rowset/segment_reader.cpp]: You should make you initialization order in accordance with variable declaration order in header file.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_reader.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/segment_reader.cpp]: yes, this modification is for this purpose to fix compile warning problems

File: be/src/olap/rowset/rowset_meta.h (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/rowset_meta.h]: ok, I will fix it.

**Pull request merged**